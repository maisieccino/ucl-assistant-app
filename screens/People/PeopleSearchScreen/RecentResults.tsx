import React from "react"
import { View } from "react-native"
import Button from "../../../components/Button"
import SearchResult from "../../../components/SearchResult"
import { CentredText, SubtitleText } from "../../../components/Typography"
import type { Person } from "../../../types/uclapi"

interface RecentResultProp {
  recent: Person,
  viewPerson: () => void,
}

const RecentResult: React.FC<RecentResultProp> = ({ recent, viewPerson }) => (recent ? (
  <SearchResult
    key={recent.email}
    topText={recent.name}
    bottomText={recent.department}
    type="person"
    onPress={viewPerson}
  />
) : null)

interface RecentResultsProps {
  viewPerson: (e: string) => () => void,
  recents: Person[],
  clearRecentResults: () => void,
}

const RecentResults: React.FC<RecentResultsProps> = ({
  recents = [],
  clearRecentResults,
  viewPerson,
}) => {
  if (recents.length === 0) {
    return null
  }
  return (
    <View>
      <SubtitleText>Recently Searched</SubtitleText>
      {recents.map((recent) => (
        <RecentResult
          key={`${recent.email}-${recent.name}`}
          recent={recent}
          viewPerson={viewPerson(recent.email)}
        />
      ))}
      {recents.length > 0 ? (
        <Button onPress={clearRecentResults}>Clear</Button>
      ) : (
        <CentredText>Recent results will appear here.</CentredText>
      )}
    </View>
  )
}

export default RecentResults
