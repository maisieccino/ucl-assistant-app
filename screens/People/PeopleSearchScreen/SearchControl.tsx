import { StackNavigationProp } from '@react-navigation/stack'
import React, { useCallback, useRef, useState } from "react"
import { ActivityIndicator, StyleSheet, View } from "react-native"
import { connect, ConnectedProps } from "react-redux"
import {
  PeopleDispatch,
  search as searchAction,
  searchClear as searchClearAction,
} from "../../../actions/peopleActions"
import { SmallButton } from "../../../components/Button"
import { Horizontal } from "../../../components/Containers"
import { TextInput } from "../../../components/Input"
import SearchResult from "../../../components/SearchResult"
import { CentredText } from "../../../components/Typography"
import type { AppStateType } from '../../../configureStore'
import type { Person } from '../../../types/uclapi'
import type { PeopleNavigatorParamList } from '../PeopleNavigator'

const styles = StyleSheet.create({
  textInput: {
    flex: 1,
    marginRight: 10,
  },
})

const MIN_QUERY_LENGTH = 4
const SEARCH_DELAY = 500

const SearchStatus = ({ query, searchResults = [] }) => {
  if (query.length === 0) {
    return <CentredText>Start typing to get search results </CentredText>
  }
  if (query.length < MIN_QUERY_LENGTH && searchResults.length === 0) {
    return <CentredText>Please enter a few more characters </CentredText>
  }
  if (query.length > 0 && searchResults.length === 0) {
    return <CentredText>No results found.</CentredText>
  }
  return null
}

interface Props extends PropsFromRedux {
  navigation: StackNavigationProp<PeopleNavigatorParamList>,
}

export const SearchControl: React.FC<Props> = ({
  search,
  token,
  clearRecentResults,
  error = ``,
  isSearching = false,
  searchResults = [],
  navigation,
}) => {
  const searchTimer = useRef<ReturnType<typeof setTimeout>>()
  const [query, setQuery] = useState(``)

  const onSearch = useCallback((q: string) => search(token, q), [search, token])
  const onQueryChange = useCallback((q: string): void => {
    clearTimeout(searchTimer.current)
    searchTimer.current = setTimeout(
      () => onSearch(q),
      SEARCH_DELAY,
    )
    setQuery(q)
  }, [onSearch, searchTimer])
  const clear = useCallback((): void => {
    clearRecentResults()
    setQuery(``)
  }, [clearRecentResults])
  const viewPerson = useCallback(
    (person: Person) => () => navigation.navigate(`PeopleDetail`, person),
    [navigation],
  )

  return (
    <View>
      <Horizontal>
        <TextInput
          placeholder="Search for a name or email..."
          onChangeText={onQueryChange}
          value={query}
          clearButtonMode="always"
          style={styles.textInput}
        />
        {
          query.length > 0 ? (
            <SmallButton onPress={clear}> Clear </SmallButton>
          ) : null
        }
      </Horizontal>
      {
        error.length > 0 && query.length > 2 && (
          <CentredText>
            {`Error! ${error} `}
            {` `}
          </CentredText>
        )
      }

      { isSearching && <ActivityIndicator /> }

      <SearchStatus query={query} searchResults={searchResults} />

      {searchResults.map((result) => (
        <SearchResult
          key={`${result.email}-${result.name}`}
          topText={result.name}
          bottomText={result.department}
          type="person"
          onPress={viewPerson(result)}
        />
      ))}
    </View>
  )
}

const connector = connect(
  (state: AppStateType) => ({
    error: state.people.searchError,
    isSearching: state.people.isSearching,
    searchResults: state.people.searchResults,
    token: state.user.token,
  }),
  (dispatch: PeopleDispatch) => ({
    clearRecentResults: () => dispatch(searchClearAction()),
    search: (
      token: string,
      query: string,
    ) => dispatch(searchAction(token, query)),
  }),
)

type PropsFromRedux = ConnectedProps<typeof connector>

export default connector(SearchControl)
