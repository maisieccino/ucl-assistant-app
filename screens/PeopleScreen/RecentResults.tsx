import React from "react"
import { View } from "react-native"
import { connect } from "react-redux"
import { generate } from "shortid"

import {
  clearRecentResults as clearRecentsResultsAction,
} from "../../actions/peopleActions"
import Button from "../../components/Button"
import SearchResult from "../../components/SearchResult"
import { CentredText, SubtitleText } from "../../components/Typography"
import type { AppStateType } from '../../configureStore'

interface Props {
  clearRecentResults: () => void,
  navigation: any,
  recents: any,
}

export class RecentResults extends React.Component<Props> {
  static mapDispatchToProps = (dispatch): any => ({
    clearRecentResults: (): any => dispatch(clearRecentsResultsAction()),
  })

  static mapStateToProps = (state: AppStateType): any => ({
    recents: state.people.recents,
  })

  static defaultProps = {
    clearRecentResults: (): void => { },
    recents: [],
  }

  viewPerson = (person) => (): void => {
    const { navigation } = this.props
    navigation.navigate(`PersonDetail`, person)
  }

  renderRecent = (recent = null) => {
    if (recent === null) {
      return null
    }
    return (
      <SearchResult
        key={generate()}
        topText={recent.name}
        bottomText={recent.department}
        type="person"
        onPress={this.viewPerson(recent)}
      />
    )
  }

  render(): React.ReactElement {
    const { recents, clearRecentResults } = this.props
    if (recents.length === 0) {
      return null
    }
    return (
      <View>
        <SubtitleText>Recently Searched</SubtitleText>
        {recents.map(this.renderRecent)}
        {recents.length > 0 ? (
          <Button onPress={clearRecentResults}>Clear</Button>
        ) : (
            <CentredText>Recent results will appear here.</CentredText>
        )}
      </View>
    )
  }
}

export default connect(
  RecentResults.mapStateToProps,
  RecentResults.mapDispatchToProps,
)(RecentResults)
