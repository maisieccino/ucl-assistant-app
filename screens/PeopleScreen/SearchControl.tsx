import React from "react"
import { ActivityIndicator, StyleSheet, View } from "react-native"
import { connect } from "react-redux"
import { generate } from "shortid"

import {
  search as searchAction,
  searchClear as searchClearAction,
} from "../../actions/peopleActions"
import { SmallButton } from "../../components/Button"
import { Horizontal } from "../../components/Containers"
import { TextInput } from "../../components/Input"
import SearchResult from "../../components/SearchResult"
import { CentredText } from "../../components/Typography"

const styles = StyleSheet.create({
  textInput: {
    flex: 1,
    marginRight: 10,
  },
})

const MIN_QUERY_LENGTH = 4

interface Props {
  clearRecentResults: () => {},
  error: any,
  isSearching: boolean,
  navigation: any,
  search: () => {},
  searchResults: any,
  token: string,
}

export class SearchControl extends React.Component<Props> {
  static mapStateToProps = (state): any => ({
    error: state.people.searchError,
    isSearching: state.people.isSearching,
    searchResults: state.people.searchResults,
    token: state.user.token,
  })

  static mapDispatchToProps = (dispatch): any => ({
    clearRecentResults: (): any => dispatch(searchClearAction()),
    search: (token, query): any => dispatch(searchAction(token, query)),
  })

  searchTimer = null

  public static defaultProps = {
    clearRecentResults: (): void => { },
    error: ``,
    isSearching: false,
    search: (): void => { },
    searchResults: [],
    token: ``,
  }

  constructor(props) {
    super(props)

    this.state = {
      query: ``,
    }
  }

  onQueryChange = (query: string): void => {
    clearTimeout(this.searchTimer)
    this.searchTimer = setTimeout(
      () => this.search(query),
      SearchControl.SEARCH_DELAY,
    )
    this.setState({ query })
  }

  search = (query: string): void => {
    const { search, token } = this.props
    search(token, query)
  }

  clear = (): void => {
    const { clearRecentResults } = this.props
    clearRecentResults()
    this.setState({ query: `` })
  }

  renderStatusText = () => {
    const { query } = this.state
    const { searchResults } = this.props
    if (query.length === 0) {
      return <CentredText>Start typing to get search results </CentredText>
    }
    if (query.length < MIN_QUERY_LENGTH && searchResults.length === 0) {
      return <CentredText>Please enter a few more characters </CentredText>
    }
    if (query.length > 0 && searchResults.length === 0) {
      return <CentredText>No results found.</CentredText>
    }
    return null
  }

  viewPerson = (person) => (): void => {
    const { navigation } = this.props
    navigation.navigate(`PersonDetail`, person)
  }

  static SEARCH_DELAY = 500

  renderResult = (res = null) => {
    if (res === null) {
      return null
    }
    return (
      <SearchResult
        key={generate()}
        topText={res.name}
        bottomText={res.department}
        type="person"
        buttonText="View"
        onPress={this.viewPerson(res)}
      />
    )
  }

  render() {
    const { query } = this.state
    const {
      error,
      isSearching,
      searchResults,
    } = this.props
    return (
      <View>
        <Horizontal>
          <TextInput
            placeholder="Search for a name or email..."
            onChangeText={this.onQueryChange}
            value={query}
            clearButtonMode="always"
            style={styles.textInput}
          />
          {
            query.length > 0 ? (
              <SmallButton onPress={this.clear}> Clear </SmallButton>
            ) : null
          }
        </Horizontal>
        {
          error.length > 0 && query.length > 2 && (
            <CentredText>
              {`Error! ${error} `}
              {` `}
            </CentredText>
          )
        }

        {
          isSearching && <ActivityIndicator />
        }

        {this.renderStatusText()}

        {searchResults.map(this.renderResult)}
      </View>
    )
  }
}

export default connect(
  SearchControl.mapStateToProps,
  SearchControl.mapDispatchToProps,
)(SearchControl)
