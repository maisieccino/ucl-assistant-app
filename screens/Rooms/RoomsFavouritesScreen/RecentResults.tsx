import { StackNavigationProp } from '@react-navigation/stack'
import React from "react"
import { View } from "react-native"
import { connect, ConnectedProps } from "react-redux"
import { generate } from "shortid"

import { clearRecents } from "../../../actions/roomsActions"
import Button from "../../../components/Button"
import SearchResult from "../../../components/SearchResult"
import { CentredText, SubtitleText } from "../../../components/Typography"
import type { AppStateType } from '../../../configureStore'
import type { Room } from '../../../types/uclapi'
import type { RoomsNavigatorParamList } from "../RoomsNavigator"

interface Props {
  navigation: StackNavigationProp<RoomsNavigatorParamList>,
}

class RecentResults extends React.Component<Props & PropsFromRedux> {
  navigateToRoomDetail = (room: Room) => () => {
    const { navigation } = this.props
    navigation.navigate(`RoomsDetail`, { room })
  }

  render() {
    const { recents, clearRecentRooms } = this.props
    if (recents.length === 0) {
      return null
    }
    return (
      <View>
        <SubtitleText>Recently Searched</SubtitleText>
        {recents.map((res) => (
          <SearchResult
            key={generate()}
            topText={res.roomname}
            bottomText={res.classification_name}
            type="location"
            onPress={this.navigateToRoomDetail(res)}
          />
        ))}
        {recents.length > 0 ? (
          <Button onPress={clearRecentRooms}>Clear</Button>
        ) : (
            <CentredText>Recent results will appear here.</CentredText>
        )}
      </View>
    )
  }
}

const connector = connect(
  (state: AppStateType) => ({
    recents: state.rooms.recents,
  }),
  (dispatch) => ({
    clearRecentRooms: () => dispatch(clearRecents()),
  }),
)

type PropsFromRedux = ConnectedProps<typeof connector>

export default connector(RecentResults)
