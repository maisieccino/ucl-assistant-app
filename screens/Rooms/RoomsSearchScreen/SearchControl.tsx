import { RouteProp } from '@react-navigation/native'
import { StackNavigationProp } from '@react-navigation/stack'
import React from "react"
import { ActivityIndicator, StyleSheet, View } from "react-native"
import { connect, ConnectedProps } from "react-redux"

import { addRecent } from "../../../actions/roomsActions"
import { SearchInput } from "../../../components/Input"
import SearchResult from "../../../components/SearchResult"
import { CentredText } from "../../../components/Typography"
import type { AppStateType } from '../../../configureStore'
import ApiManager from "../../../lib/ApiManager"
import type { RoomsNavigatorParamList } from "../RoomsNavigator"

const styles = StyleSheet.create({
  container: {
    paddingBottom: 20,
  },
})

const MIN_QUERY_LENGTH = 4
const SEARCH_DELAY = 500

interface Props extends PropsFromRedux {
  navigation: StackNavigationProp<RoomsNavigatorParamList>,
  // eslint-disable-next-line quotes
  route: RouteProp<RoomsNavigatorParamList, 'RoomsSearch'>,
}

interface State {
  error: Error,
  isSearching: boolean,
  query: string,
  searchResults: Array<unknown>,
}

class SearchControl extends React.Component<Props, State> {
  private subscriptions

  private searchTimer

  constructor(props) {
    super(props)
    this.state = {
      error: null,
      isSearching: false,
      query: ``,
      searchResults: [],
    }
  }

  componentDidMount() {
    const { navigation } = this.props
    this.subscriptions = [
      navigation.addListener(`focus`, this.componentDidFocus),
    ]
    this.componentDidFocus()
  }

  componentWillUnmount() {
    this.subscriptions.forEach((sub) => {
      if (sub && typeof sub.remove === `function`) {
        sub.remove()
      }
    })
  }

  componentDidFocus = () => {
    const { route } = this.props
    const { query } = this.state
    const queryExists = this.queryExists(route)
    if (queryExists && query !== route.params.query) {
      this.setState({ query: route.params.query })
      this.searchRooms(route.params.query, true)
    }
  }

  queryExists = (route) => route
    && route.params
    && route.params.query
    && route.params.query.length > 0

  onChangeText = (query: string) => {
    if (query.length >= MIN_QUERY_LENGTH) {
      clearTimeout(this.searchTimer)
      this.searchTimer = setTimeout(
        () => this.searchRooms(query),
        SEARCH_DELAY,
      )
    }
    this.setState({ query })
  }

  searchRooms = async (query: string, autoNavigate = false) => {
    const { token } = this.props
    try {
      this.setState({ isSearching: true })
      const results = await ApiManager.rooms.search(token, query)
      if (autoNavigate && results.length === 1) {
        this.navigateToRoomDetail(results[0])()
      }
      this.setState({ isSearching: false, searchResults: results })
    } catch (error) {
      this.setState({ error: error.message, isSearching: false })
    }
  }

  navigateToRoomDetail = (room) => () => {
    const { navigation, addRecentRoom } = this.props
    addRecentRoom(room)
    navigation.navigate(`RoomsDetail`, { room })
  }

  clear = () => this.setState({ query: ``, searchResults: [] })

  renderSearchResult = (searchResult) => (
    <SearchResult
      key={searchResult.roomid}
      topText={searchResult.roomname}
      bottomText={searchResult.classification_name}
      type="location"
      onPress={this.navigateToRoomDetail(searchResult)}
    />
  )

  renderStatusText = () => {
    const { query, searchResults } = this.state
    if (query.length === 0) {
      return <CentredText>Start typing to get search results</CentredText>
    }
    if (query.length < MIN_QUERY_LENGTH && searchResults.length === 0) {
      return <CentredText>Please enter a few more characters</CentredText>
    }
    if (query.length > 0 && searchResults.length === 0) {
      return <CentredText>No results found.</CentredText>
    }
    return null
  }

  render() {
    const {
      query, error, isSearching, searchResults,
    } = this.state
    return (
      <View style={styles.container}>
        <SearchInput
          placeholder="Search for a room or building name..."
          onChangeQuery={this.onChangeText}
          query={query}
          clear={this.clear}
        />
        {error ? (
          <CentredText>
            {`Error! ${error} `}
          </CentredText>
        ) : null}
        {isSearching ? <ActivityIndicator /> : null}
        {this.renderStatusText()}
        {searchResults.map(this.renderSearchResult)}
      </View>
    )
  }
}

const connector = connect(
  (state: AppStateType) => ({
    token: state.user.token,
  }),
  (dispatch) => ({
    addRecentRoom: (room) => dispatch(addRecent(room)),
  }),
)

type PropsFromRedux = ConnectedProps<typeof connector>

export default connector(SearchControl)
