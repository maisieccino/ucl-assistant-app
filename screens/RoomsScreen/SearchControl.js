// @flow
import PropTypes from "prop-types"
import React, { Component } from "react"
import { ActivityIndicator, StyleSheet, View } from "react-native"
import { connect } from "react-redux"

import { addRecent } from "../../actions/roomsActions"
import { SearchInput } from "../../components/Input"
import SearchResult from "../../components/SearchResult"
import { CentredText } from "../../components/Typography"
import ApiManager from "../../lib/ApiManager"

const styles = StyleSheet.create({
  container: {
    paddingBottom: 20,
  },
})

const MIN_QUERY_LENGTH = 4
const SEARCH_DELAY = 500

class SearchControl extends Component {
  static mapStateToProps = (state) => ({
    token: state.user.token,
  })

  static mapDispatchToProps = (dispatch) => ({
    addRecentRoom: (room) => dispatch(addRecent(room)),
  })

  static propTypes = {
    addRecentRoom: PropTypes.func,
    navigation: PropTypes.shape().isRequired,
    token: PropTypes.string,
  }

  static defaultProps = {
    addRecentRoom: () => { },
    token: ``,
  }

  constructor(props) {
    super(props)
    this.state = {
      error: null,
      isSearching: false,
      query: ``,
      searchResults: [],
    }
  }

  componentDidMount() {
    const { navigation, route } = this.props
    this.subscriptions = [
      navigation.addListener(`focus`, this.componentDidFocus),
    ]
    this.componentDidFocus({ state: route })
  }

  componentWillUnmount() {
    this.subscriptions.forEach((sub) => sub.remove())
  }

  componentDidFocus = (route) => {
    const { query } = this.state
    const queryExists = this.queryExists(route)
    if (queryExists && query !== route.params.query) {
      this.setState({ query: route.params.query })
      this.searchRooms(route.params.query, true)
    }
  }

  queryExists = (route) => route
    && route.params
    && route.params.query
    && route.params.query.length > 0

  onChangeText = (query: string) => {
    if (query.length >= MIN_QUERY_LENGTH) {
      clearTimeout(this.searchTimer)
      this.searchTimer = setTimeout(
        () => this.searchRooms(query),
        SEARCH_DELAY,
      )
    }
    this.setState({ query })
  }

  searchRooms = async (query: string, autoNavigate = false) => {
    const { token } = this.props
    try {
      this.setState({ isSearching: true })
      const results = await ApiManager.rooms.search(token, query)
      if (autoNavigate && results.length === 1) {
        this.navigateToRoomDetail(results[0])()
      }
      this.setState({ isSearching: false, searchResults: results })
    } catch (error) {
      this.setState({ error: error.message, isSearching: false })
    }
  }

  navigateToRoomDetail = (room) => () => {
    const { navigation, addRecentRoom } = this.props
    addRecentRoom(room)
    navigation.navigate(`RoomDetail`, { room })
  }

  clear = () => this.setState({ query: ``, searchResults: [] })

  renderSearchResult = (searchResult) => (
    <SearchResult
      key={searchResult.roomid}
      topText={searchResult.roomname}
      bottomText={searchResult.classification_name}
      type="location"
      buttonText="View"
      onPress={this.navigateToRoomDetail(searchResult)}
    />
  )

  renderStatusText = () => {
    const { query, searchResults } = this.state
    if (query.length === 0) {
      return <CentredText>Start typing to get search results</CentredText>
    }
    if (query.length < MIN_QUERY_LENGTH && searchResults.length === 0) {
      return <CentredText>Please enter a few more characters</CentredText>
    }
    if (query.length > 0 && searchResults.length === 0) {
      return <CentredText>No results found.</CentredText>
    }
    return null
  }

  render() {
    const {
      query, error, isSearching, searchResults,
    } = this.state
    return (
      <View style={styles.container}>
        <SearchInput
          placeholder="Search for a room or building name..."
          onChangeQuery={this.onChangeText}
          query={query}
          clear={this.clear}
        />
        {error ? (
          <CentredText>
            {`Error! ${error} `}
          </CentredText>
        ) : null}
        {isSearching ? <ActivityIndicator /> : null}
        {this.renderStatusText()}
        {searchResults.map(this.renderSearchResult)}
      </View>
    )
  }
}

export default connect(
  SearchControl.mapStateToProps,
  SearchControl.mapDispatchToProps,
)(SearchControl)
