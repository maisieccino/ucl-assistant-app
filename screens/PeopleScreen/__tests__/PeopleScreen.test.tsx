import "react-native"

import React from "react"
import { cleanup, render } from 'react-native-testing-library'

import { RecentResults } from "../RecentResults"
import { SearchControl } from "../SearchControl"

describe(`RecentResults`, () => {
  afterEach(() => {
    cleanup()
  })

  it(`renders an empty RecentResults component`, async () => {
    const props = {
      clearRecents: jest.fn(),
      navigation: {
        navigate: jest.fn(),
      },
      recents: [],
    }
    const tree = render(<RecentResults {...props} />).toJSON()
    expect(tree).toBe(null)
  })

  it(`renders a filled RecentResults component`, async () => {
    const props = {
      clearRecents: jest.fn(),
      navigation: {
        navigate: jest.fn(),
      },
      recents: [
        {
          department: `Department of Agriculture`,
          name: `Mr Potato`,
        },
        {
          department: `Faculty of Medical Sciences`,
          name: `Chris P Bacon`,
        },
        {
          department: `Philosophy Department`,
          name: `Jeremy Bentham`,
        },
      ],
    }
    const tree = render(<RecentResults {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

it(`renders a SearchControl component`, async () => {
  const props = {
    clear: jest.fn(),
    error: ``,
    isSearching: false,
    navigation: {
      navigate: jest.fn(),
    },
    search: jest.fn(),
    searchResults: [],
    token: ``,
  }
  const component = render(<SearchControl {...props} />)
  expect(component).toMatchSnapshot()
})
