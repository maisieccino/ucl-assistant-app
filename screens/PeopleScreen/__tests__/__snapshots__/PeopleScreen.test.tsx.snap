// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentResults renders a filled RecentResults component 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#4D5061",
          "fontFamily": "apercu-bold",
          "fontSize": 28,
          "marginBottom": 2,
          "marginTop": 8,
        },
        undefined,
      ]
    }
  >
    Recently Searched
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "marginBottom": 10,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1845,
            "shadowRadius": 1.62,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#eaeaea",
            "borderRadius": 80,
            "marginRight": 10,
            "padding": 10,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4D5061",
                "fontFamily": "apercu-bold",
                "fontSize": 16,
                "marginBottom": 0,
                "marginTop": 1,
              },
              undefined,
            ]
          }
        >
          Mr Potato
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#6E707F",
                "fontFamily": "apercu",
                "fontSize": 16,
                "fontStyle": "italic",
                "marginBottom": 1,
                "marginTop": 0,
              },
              undefined,
            ]
          }
        >
          Department of Agriculture
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "marginBottom": 10,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1845,
            "shadowRadius": 1.62,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#eaeaea",
            "borderRadius": 80,
            "marginRight": 10,
            "padding": 10,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4D5061",
                "fontFamily": "apercu-bold",
                "fontSize": 16,
                "marginBottom": 0,
                "marginTop": 1,
              },
              undefined,
            ]
          }
        >
          Chris P Bacon
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#6E707F",
                "fontFamily": "apercu",
                "fontSize": 16,
                "fontStyle": "italic",
                "marginBottom": 1,
                "marginTop": 0,
              },
              undefined,
            ]
          }
        >
          Faculty of Medical Sciences
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "marginBottom": 10,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1845,
            "shadowRadius": 1.62,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#eaeaea",
            "borderRadius": 80,
            "marginRight": 10,
            "padding": 10,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4D5061",
                "fontFamily": "apercu-bold",
                "fontSize": 16,
                "marginBottom": 0,
                "marginTop": 1,
              },
              undefined,
            ]
          }
        >
          Jeremy Bentham
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#6E707F",
                "fontFamily": "apercu",
                "fontSize": 16,
                "fontStyle": "italic",
                "marginBottom": 1,
                "marginTop": 0,
              },
              undefined,
            ]
          }
        >
          Philosophy Department
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4279998859,
            4283151357,
          ],
          "endPoint": Array [
            1,
            0,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            1,
          ],
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 8,
            "elevation": 2,
            "marginBottom": 5,
            "marginTop": 5,
            "minHeight": 55,
            "minWidth": 150,
            "padding": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#1B998B",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 3,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fefefe",
              "fontFamily": "apercu",
              "fontSize": 20,
              "marginBottom": 1,
              "marginTop": 1,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Clear
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</View>
`;

exports[`renders a SearchControl component 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Search for a name or email..."
      placeholderTextColor="#6E707F"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#eaeaea",
            "borderColor": "#6E707F",
            "borderRadius": 10,
            "color": "#4D5061",
            "fontFamily": "apercu",
            "fontSize": 16,
            "height": 50,
            "marginBottom": 10,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 10,
            "minWidth": 200,
            "paddingBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
          },
          Object {
            "flex": 1,
            "marginRight": 10,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#6E707F",
          "fontFamily": "apercu",
          "fontSize": 16,
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Start typing to get search results 
  </Text>
</View>
`;
